hoomd/box.py

```diff
@@ -0,0 +1,393 @@
+import numpy as np
+from functools import partial
+import hoomd._hoomd as _hoomd
+
+
+def make_three_vec(vec, func, scalar_conversion=None, is2D=False, name=None):
+    if scalar_conversion is None:
+        def sc(x):
+            return x
+    else:
+        sc = scalar_conversion
+    if len(vec) == 2 and is2D:
+        return func(sc(vec[0]), sc(vec[1]), sc(0))
+    elif len(vec) == 3:
+        return func(sc(vec[0]), sc(vec[1]), sc(vec[2]))
+    else:
+        if name is None:
+            name = "vector"
+        if not is2D and len(vec) == 2:
+            raise ValueError("Box is 3 dimensions, but {name} was length 2. "
+                             "Give a 3 dimensional vector for {name}."
+                             "".format(name=name))
+        else:
+            raise ValueError("Length of the {} must be 2 or 3 dimensions."
+                             "".format(name))
+
+
+make_scalar3 = partial(make_three_vec, func=_hoomd.make_scalar3,
+                       scalar_conversion=float)
+
+make_int3 = partial(make_three_vec, func=_hoomd.make_int3,
+                    scalar_conversion=int)
+
+make_uint3 = partial(make_three_vec, func=_hoomd.make_uint3,
+                     scalar_conversion=int)
+
+make_char3 = partial(make_three_vec, func=_hoomd.make_char3,
+                     scalar_conversion=int)
+
+
+def to_three_array(vec):
+    return np.array((vec.x, vec.y, vec.z))
```

Comment:
Should we use tuples here to indicate that the values are not modifiable?

Reply:
I am returning NumPy arrays currently for their broadcasting and other niceties in case someone wants to do array calculations on the values given. I could see an argument for tuples though. On the other hand we could just document that all attributes return copies of the values and not the values themselves.

Reply:
I'm fine with either, as long as it is documented. I just mentioned it because we were careful to return tuples in other areas.

Reply:
I think in most of those cases we wouldn't expect the user to attempt to use the value as an array. (I could be wrong there though).

Reply:
Freud returns box vectors as numpy arrays.